### ISSUE TITLE
Temporary clone name

### ISSUE THREAD

**Issue by rmillwood** [jmoenig/Snap#2713]
Snap! Reference Manual v6 p62 says in G. List of attributes that  "name: my name (same as parent’s name if I’m temporary)" but I can only see empty names. 

Would it be useful if temporary clone names were unique e.g. <parentname>(2), <parentname>(3) etc ?

I am working on a simulation using temporsary clones, but struggling to find a way for them to identify themselves.

**Comment by brianharvey** [comment]
The usual thing is to have a sprite-local variable ID and increment it before creating each clone.  So each clone starts with a different value of ID.

**Comment by rmillwood** [comment]
Yes - I am getting there slowly! :) Would this also allow one to directly reference the clone?

Nevertheless, the name is not set at all at present, and if it where as the manual suggest, what value would there be in each clone having the same name as the parent?

**Comment by jguille2** [comment]
Hi Richard @rmillwood ,

Some comments... maybe they help you.

- When you create clons, you can do a lot of things... for example, you can set their names.

![clone1](https://user-images.githubusercontent.com/2926315/98553181-a0ed9000-229f-11eb-934f-760655f9d042.png)

- And then, you can talk with this clones by name

![clone2](https://user-images.githubusercontent.com/2926315/98553366-d3978880-229f-11eb-8393-2525fcf1d6ba.png)

- But names are not the only strategy. For example, you can remember them inside a variable (first class data don't need names or Ids everywhere). For example, using a list:
![clone3](https://user-images.githubusercontent.com/2926315/98553652-22452280-22a0-11eb-8246-48dadc9ec202.png)

- And then, just use your directly that list:
![clone4](https://user-images.githubusercontent.com/2926315/98553759-3ee15a80-22a0-11eb-8480-89c744281410.png)

Joan

**Comment by rmillwood** [comment]
Thanks Joan - very helpful. 
I learnt a lot about code sequence and clones - here is the latest draft: http://snap.berkeley.edu/run#present:Username=rmillwood&ProjectName=Century%20Club
